homeassistant/components/starlink/coordinator.py

```diff
@@ -119,12 +119,16 @@ async def async_set_sleep_start(self, start: int) -> None:
 
     async def async_set_sleep_duration(self, end: int) -> None:
         """Set Starlink system sleep schedule end time."""
+        duration = end - self.data.sleep[0]
```

Comment:
We shouldn't try to do datetime math ourselves. Use datetime objects and than convert back the `total_seconds` when done

Reply:
I mentioned it in another comment above, but I'm fairly sure we are dealing with "seconds since 00:00" in this case.
If that is indeed the case, using datetime objects will actually cause issues, especially around daylight savings.
If that's not the case, I'm happy to change this :slightly_smiling_face:

Reply:
From the doc string it looks like everything is in UTC so you should be able to use UTC time which doesn't have timezone or DST concerns

https://github.com/sparky8512/starlink-grpc-tools/blob/37aeed63defcc9a8c3f52c9d8c7607988fede1c5/starlink_grpc.py#L1615

Reply:
Apologies for the delay, I was working up the courage to fight Python again 😅
I've been poking around in docs, and it doesn't look like we can actually add/subtract `datetime.time` instances from each other. Were you proposing that we concern ourselves with dates here? That seems redundant.

Reply:
It's a bit convoluted that it gets converted in `_time_to_utc_minutes` and then converted again in `async_set_sleep_duration` function.  It would be nicer to convert it once, but you have another function in that case, so on second thought, I think this is fine.